grid_layout.go

```diff
@@ -0,0 +1,12 @@
+package gxui
+
+type GridLayout interface {
```

Comment:
I was discussing this CL with one of my colleagues, and we were thinking that perhaps TableLayout is a better name given that it has similarities to HTML's tables - in that you can make children span multiple cells. I was also planning on adding GridList support, and I'm weary of overloading Grid.